File: docs/sdk/council_api_types.py
"""Generated type definitions for ai-decision-council API."""

from __future__ import annotations

from typing import Any, TypedDict


class ApiError(TypedDict):
    code: str
    message: str


class ApiEnvelope(TypedDict):
    data: Any | None
    metadata: dict[str, Any]
    errors: list[ApiError]


class ConversationSummary(TypedDict):
    id: str
    created_at: str
    title: str
    message_count: int


class Conversation(TypedDict):
    id: str
    created_at: str
    title: str
    messages: list[dict[str, Any]]


class AssistantMessage(TypedDict, total=False):
    id: str
    role: str
    stage1: list[dict[str, Any]]
    stage2: list[dict[str, Any]]
    stage3: dict[str, Any]
    metadata: dict[str, Any]
    errors: list[ApiError]
    created_at: str


class SendMessageData(TypedDict, total=False):
    conversation: Conversation
    assistant_message: AssistantMessage
    result: dict[str, Any]


class StreamEvent(TypedDict):
    event_type: str
    event_id: str | None
    envelope: ApiEnvelope

File: docs/sdk/councilApiTypes.ts
// Generated type definitions for ai-decision-council API.

export interface ApiError {
  code: string;
  message: string;
}

export interface ApiEnvelope<TData = unknown> {
  data: TData | null;
  metadata: Record<string, unknown>;
  errors: ApiError[];
}

export interface ConversationSummary {
  id: string;
  created_at: string;
  title: string;
  message_count: number;
}

export interface Conversation {
  id: string;
  created_at: string;
  title: string;
  messages: Record<string, unknown>[];
}

export interface AssistantMessage {
  id?: string;
  role?: string;
  stage1?: Record<string, unknown>[];
  stage2?: Record<string, unknown>[];
  stage3?: Record<string, unknown>;
  metadata?: Record<string, unknown>;
  errors?: ApiError[];
  created_at?: string;
}

export interface SendMessageData {
  conversation?: Conversation;
  assistant_message?: AssistantMessage;
  result?: Record<string, unknown>;
}

export interface StreamEvent<TData = unknown> {
  eventType: string;
  eventId: string | null;
  envelope: ApiEnvelope<TData>;
}
